### ISSUE TITLE
patternCircular3d doesn't work with negative extrudes

### ISSUE THREAD

**Issue by jgomez720** [KittyCAD/modeling-app#1990]
## Description
When using `patternCircular3d` with a negative extrude, the extrusion no longer removes material.

## Recording
https://github.com/KittyCAD/modeling-app/assets/114548659/a63c67e2-48bf-4d94-a58d-f5f2ee5985d1

### Version
v0.17.1

**Comment by jgomez720** [comment]
Once again, not sure who this belongs to. I only picked you @gserena01 because you are now the pattern queen. Please throw to whoever it belongs to

**Comment by jessfraz** [comment]
paste the code :)

**Comment by jgomez720** [comment]
thank you for reminding me. I changed up the code a bit from the video, but this still produces the same issue
```
const flangeThickness = .125
const flangeBaseDia = 1.25
const boreHeight = 1
const flangePipeDia = 1
const mountingHoleDia = 0.375

const flange = startSketchOn('XY')
  |> startProfileAt([flangePipeDia / 2, boreHeight], %)
  |> line([flangeThickness, 0], %)
  |> line([0, -boreHeight], %)
  |> line([flangeBaseDia, 0], %, 'revolveSketch')
  |> line([0, -flangeThickness], %)
  |> line([-flangeBaseDia - flangeThickness, 0], %)
  |> close(%)
  |> revolve({ axis: 'y' }, %)

const holesOnFlange = startSketchOn(flange, 'revolveSketch')
  |> circle([.2, 0], mountingHoleDia / 2, %)
  |> extrude(-flangeThickness, %)
  |> patternCircular3d({
  axis: [0, 1, 0],
  center: [0, 0, 0],
  repetitions: 6,
  arcDegrees: 360,
  rotateDuplicates: true
  }, %)
```

**Comment by gserena01** [comment]
This looks like it's more of an engine-side issue. I'll take a look later today

**Comment by gserena01** [comment]
@jgomez720 does this issue get recreated with patternLinear3d?

**Comment by jgomez720** [comment]
TL;DR - the issue does not get recreated with `patternLinear3d`
https://github.com/KittyCAD/modeling-app/assets/114548659/3056f742-a48c-482f-aa36-a54c19254ffc

The other issue I am mentioning in the video is https://github.com/KittyCAD/engine/issues/1905

**Comment by jgomez720** [comment]
Discovered that this is not a separate issue from https://github.com/KittyCAD/engine/issues/1905. In the flange example, the entire flange is being patterned, so it's covering the holes, making it look like they are no longer negative extrudes
